File: examples/typescript/posts.mjs
import { EnvoAPI } from 'envoapi';

const client = new EnvoAPI({ baseUrl: process.env.ENVOAPI_BASE_URL });
const result = await client.profiles.getPosts({ username: process.env.ENVOAPI_USERNAME ?? 'alice' });
console.log(JSON.stringify({posts: result.body.data.posts, creditCost: result.body.meta.creditCost}));

File: examples/python/posts.py
import os
from envoapi import EnvoAPI

with EnvoAPI(
    base_url=os.environ.get("ENVOAPI_BASE_URL", "https://api.envoapi.com")
) as client:
    result = client.profiles.get_posts(
        username=os.environ.get("ENVOAPI_USERNAME", "alice")
    )
    print(result.body.to_dict())
